sycl/source/detail/queue_impl.cpp

```diff
@@ -35,33 +37,48 @@ template <> device queue_impl::get_info<info::queue::device>() const {
 
 event queue_impl::memset(void *Ptr, int Value, size_t Count) {
   context Context = get_context();
-  std::shared_ptr<usm::USMDispatcher> USMDispatch =
-      getSyclObjImpl(Context)->getUSMDispatch();
-  cl_event Event;
+  if (Context.is_host()) {
+    std::memset(Ptr, Value, Count);
```

Comment:
Having implementation of this function doesn't look right to me. I would not expect that queue object does these operations with memory. What about creating/reusing dedicated function in memory_manager for these stuff?

Reply:
This is meant as a convenience function for the cases where you just want this done now without any dependences blocking the operation.  What you you suggest seems like it would be more complex.

Reply:
I'm not against having memset method in queue class, I just don't think actual work should be done in this function. I suggest reusing function of MemoryManager.

Reply:
I tried, but it's too complicated.  Memory_Manager interfaces take a QueueImplPtr, and modifying the queue_impl to be able to return a shared_ptr to itself is frankly not worth the effort.

Reply:
You can additionally pass QueueImplPtr from queue like we do in other methods, so this function will look like:
```
event queue_impl::memset(shared_ptr<queue_impl> Impl, void *Ptr, int Value, size_t Count) {
  cl_event Event = nullptr;
  MemoryManager::fill_usm(Ptr, Impl, Count, Value, /*DepEvents*/ {}, Event);
  return event(Event, Context);
}
```

Reply:
...d'oh.

Yes, you're right.